src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartDataGroupingsForm.tsx

```diff
@@ -0,0 +1,230 @@
+import ChartCustomDataGroupingsConfiguration from '@admin/pages/release/datablocks/components/chart/ChartCustomDataGroupingsConfiguration';
+import styles from '@admin/pages/release/datablocks/components/chart/ChartDataGroupingsForm.module.scss';
+import Button from '@common/components/Button';
+import ButtonGroup from '@common/components/ButtonGroup';
+import { Form, FormFieldRadioGroup, FormSelect } from '@common/components/form';
+import FormFieldNumberInput from '@common/components/form/FormFieldNumberInput';
+import {
+  CustomDataGroup,
+  DataGrouping,
+  DataGroupingType,
+} from '@common/modules/charts/types/chart';
+import expandDataSet from '@common/modules/charts/util/expandDataSet';
+import generateDataSetKey from '@common/modules/charts/util/generateDataSetKey';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import { OmitStrict } from '@common/types';
+import Yup from '@common/validation/yup';
+import { Formik } from 'formik';
+import orderBy from 'lodash/orderBy';
+import React from 'react';
+import generateDataSetLabel from './utils/generateDataSetLabel';
+
+interface FormValues extends OmitStrict<DataGrouping, 'type'> {
+  numberOfGroupsQuantiles?: number;
+  copyCustomGroups?: string;
+  type: DataGroupingType | 'CopyCustom';
+}
+
+interface Props {
+  dataGrouping: DataGrouping;
+  dataSetsWithCustomGroups: DataGrouping[];
+  id: string;
+  meta: FullTableMeta;
+  unit?: string;
```

Comment:
This is currently unused (so could be removed), but as mentioned in another comment, we could actually implement this to clean up the `DataGrouping` type (which unnecessarily has `unit` as a property).

For example, one way we could do this:

```tsx
// ChartDataGroupingsConfiguration.tsx

interface DataGroupingWithUnit extends DataGrouping {
  unit: string;
}

interface FormValues {
  dataGroupings: DataGroupingWithUnit[];
}

const ChartDataGroupingsConfiguration = (...) => {
  const [editGrouping, setEditGrouping] = useState<DataGroupingWithUnit>();
  
  return (
    <ChartDataGroupingsForm
      dataGrouping={editGrouping}
      unit={editGrouping.unit} 
    />
  );
}
```

Reply:
Couldn't easily see if this was resolved with a new interface - was it handled another way?

Reply:
yep, I changed it to get the unit in `ChartDataGroupingsConfiguration` and pass it down to `ChartDataGroupingForm` rather than add it to the `DataGrouping` type.